tools/castle-editor/code/castleundosystem.pas

```diff
@@ -0,0 +1,276 @@
+{ An undo system that handles:
+  Recording the changes made to the designed Scene
+  Performing Undo/Redo actions based on current undo history }
+unit CastleUndoSystem;
+
+{$mode objfpc}{$H+}
+
+interface
+
+uses
+  Classes, Generics.Collections,
+  CastleUtils;
+
+type
+  { Content of the Undo record
+    we might want to change it in future, most likely make several types of data
+    that can be recorded as undo events }
+  TUndoData = string;
```

Comment:
It should be `String` (`CamelCase`) :) We're trying to follow this in new CGE code.

Reply:
Done!